src/pages/signin/EmailDeliveryFailurePage.js

```diff
@@ -0,0 +1,96 @@
+import React, {useEffect} from 'react';
+import {Keyboard, View} from 'react-native';
+import {withOnyx} from 'react-native-onyx';
+import PropTypes from 'prop-types';
+import Str from 'expensify-common/lib/str';
+import styles from '../../styles/styles';
+import Text from '../../components/Text';
+import TextLink from '../../components/TextLink';
+import ONYXKEYS from '../../ONYXKEYS';
+import useLocalize from '../../hooks/useLocalize';
+import useKeyboardState from '../../hooks/useKeyboardState';
+import redirectToSignIn from '../../libs/actions/SignInRedirect';
+import CONST from '../../CONST';
+import PressableWithFeedback from '../../components/Pressable/PressableWithFeedback';
+
+const propTypes = {
+    /* Onyx Props */
+
+    /** The credentials of the logged in person */
+    credentials: PropTypes.shape({
+        /** The email/phone the user logged in with */
+        login: PropTypes.string,
+    }),
+};
+
+const defaultProps = {
+    credentials: {},
+};
+
+function EmailDeliveryFailurePage(props) {
+    const {isKeyboardShown} = useKeyboardState();
+    const {translate} = useLocalize();
+    const login = Str.isSMSLogin(props.credentials.login) ? Str.removeSMSDomain(props.credentials.login) : props.credentials.login;
+
+    useEffect(() => {
+        if (!isKeyboardShown) {
+            return;
+        }
+        Keyboard.dismiss();
+    }, [isKeyboardShown]);
```

Comment:
Let's add a comment why this was added, as it look unnecessary here

Reply:
Done!